From 74a606c8348ab195bdc4573d88d9df2f35654b18 Mon Sep 17 00:00:00 2001
From: Neha Bisht <quic_nbisht@quicinc.com>
Date: Thu, 6 Oct 2022 14:37:49 +0530
Subject: [PATCH] skb_recycler: Add flags in skb

Add flags in skb for checking if skb is fast recycled and
packet is coming from recycler

Change-Id: Iea91f75a7989799da71006b4609ba7326c9e07cd
Signed-off-by: Neha Bisht <quic_nbisht@quicinc.com>
---
 include/linux/skbuff.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index 21a94287edfe..bade53e3b92d 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -989,6 +989,10 @@ struct sk_buff {
 	__u8			fast_forwarded:1;
 	/* Linear packets processed by dev_fast_xmit() */
 	__u8			fast_xmit:1;
+	/* Flag to check if skb is allocated from recycler */
+	__u8			is_from_recycler:1;
+	/* Flag for fast recycle in fast xmit path */
+	__u8			fast_recycled:1;
 	/* 1 or 3 bit hole */
 
 #ifdef CONFIG_NET_SCHED
-- 
2.34.1

